server.go

```diff
@@ -1316,6 +1320,32 @@ func (s *Server) Authenticate(username, password string) (*User, error) {
 	return u, nil
 }
 
+// Authenticate by secret file
+func (s *Server) AuthenticateBySecret(secret string) (*User, error) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	// load content of the file (first line only)
+	file, err := os.Open(s.SecretFile)
```

Comment:
Basically I think when the server starts up, it should know if the secret file config has been provided.  If so, we should simply feed the secret to it as a startup param.  All the checking of the file, valid content of the file, etc can be done when parsing the config and then appropriately loading the file.  Then `secret` just becomes a field in the `server` struct.